src/pwrite.rs

```diff
@@ -24,12 +24,11 @@ use crate::error;
 /// have a data that is not `&[u8]` — you will probably want to implement
 /// [TryIntoCtx](ctx/trait.TryIntoCtx.html) on your Rust types to be written.
 ///
-pub trait Pwrite<Ctx, E>: Index<usize> + IndexMut<RangeFrom<usize>> + MeasureWith<Ctx>
-where
-    Ctx: Copy,
-    E: From<error::Error>,
-{
-    fn pwrite<N: TryIntoCtx<Ctx, <Self as Index<RangeFrom<usize>>>::Output, Error = E>>(
+pub trait Pwrite<Ctx: Copy, E> {
```

Comment:
Kind of talking to myself: So re my worries about dispatch ambiguity, i think it's not possilble; since implementors (e.g., blanket impl + DynamicBuffer) implement Pwrite, etc., and so disaptch is always `thing_that_implements_pwrite.pwrite` so there won't be any dispatch ambiguity, so yea that's a worry I don't need to have.